spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -223,8 +233,34 @@ else if (isCountingQuery && this.tree.isDelete()) {
 		}
 	}
 
+	private Supplier<Builder<?>> getQueryBuilderSupplier(Object[] parameters) {
+		Supplier<Builder<?>> queryBuilderSupplier;
+		queryBuilderSupplier = StructuredQuery::newEntityQueryBuilder;
+		ProjectionInformation projectionInformation =
+				this.projectionFactory.getProjectionInformation(this.queryMethod.getReturnedObjectType());
+
+		if (projectionInformation != null &&
```

Comment:
In what cases would the returned `projectInformation` be null?

Reply:
it happens in some tests